src/Debugger/Breakpoint.php

```diff
@@ -548,30 +581,91 @@ public function validate()
             );
             return false;
         }
+        return true;
+    }
+
+    private function validateSourceLocation()
+    {
+        if (!$this->location) {
+            $this->setError(
+                StatusMessage::REFERENCE_BREAKPOINT_SOURCE_LOCATION,
+                'Invalid breakpoint location'
+            );
+            return false;
+        }
+
+        $path = $this->location->path();
+        $info = new \SplFileInfo($path);
+
+        // Ensure the file exists and is readable
+        if (!$info->isReadable()) {
+            $this->setError(
+                StatusMessage::REFERENCE_BREAKPOINT_SOURCE_LOCATION,
+                'Invalid breakpoint location - File not found or unreadable: $0.',
+                [$path]
+            );
+            return false;
+        }
 
-        if ($this->condition()) {
-            // validate that the condition is ok for debugging
-            try {
-                if (!stackdriver_debugger_valid_statement($this->condition())) {
-                    $this->setError(
-                        StatusMessage::REFERENCE_BREAKPOINT_CONDITION,
-                        'Invalid breakpoint condition - Invalid operations: $0.',
-                        [$this->condition]
-                    );
-                    return false;
-                }
-            } catch (\ParseError $e) {
+        // Ensure the file is a php file
+        if ($info->getExtension() !== "php") {
```

Comment:
Should we support uppercase extension (.PHP)?

Reply:
Sure, we can make this case-insensitive.